pkg/querier/chunk_merge_iterator.go

```diff
@@ -32,112 +32,120 @@ func newChunkMergeIterator(cs []chunk.Chunk) storage.SeriesIterator {
 	}
 	sort.Sort(byFrom(chunks))
 
+	// Build a list of lists of non-overlapping chunks.
```

Comment:
Consider breaking all this code out into a function that describes what it is doing instead of needing comments. It looks like the only thing used later is `its`. Perhaps something like `buildIterators(chunks)`